### DISCUSSION TITLE
Adding Return Type Polymorphism?

### DISCUSSION THREAD

**Discussion by simonsan** [rust-unofficial/patterns#254] [Ideas]
Do you think it's valuable to add something about Return Type Polymorphism to the idioms section?

EDIT: Or develop something ourselves out of it to show the (dis-)advantages in certain situations and when (not) to use it?

https://loige.co/rust-shenanigans-return-type-polymorphism/

**Comment by pickfire** [comment]
I didn't even know it can be considered an idiom, I thought it is just one of the possible usage. Isn't it too basic to be shown as an idiom?

**Reply by simonsan** [reply]
It's really basic that's true. I thought it might be nice for `rustlings`, so not really considering it an `idiom` as we already have them. But I didn't find any other term for it.

Is it well-documented in the docs, do you know it? (didn't make research myself for now)
Otherwise we could develop something of it, where to use, where not etc.

**Reply by pickfire** [reply]
I know it can be done but I never get to use it. We can probably pluck out a few examples from standard library. I believe `Default::default` and `Iterator::collect` (from the article) as well as `Into::into` are examples of this.
